CONTRIBUTION.md

```diff
@@ -28,58 +28,126 @@ We know that as of 1.12 this is not the case for great parts of the mod, but we
-    * An Suppression should always have a comment stating why it is valid to suppress this warning here
-    * In general only a `@SupressWarning("unchecked")` should ever be necessary - do not suppress deprecations!
-* Capitalize comments
-* Use multiline comments where applicable, and single-line-comments only for single line comments
-* Write Javadoc instead of comments, where possible. 
-    * `@implNote` and `@implSpec` are always possible...
-    * try to list all unchecked exceptions with `@throws`
-    * leave uncommented parameters out (no `@param obj` without description)
-* avoid giant Method-Bodies where possible, by splitting into smaller Methods
+  * In most cases this can be avoided by giving the object that's being tested on an abstract method to be overridden. For example, give Item and Block objects a method called `getGuiElement(EntityPlayer, int, int ,int)` and call it in `GuiProxy` instead of using a case statement.
+  * The other approach would be using a `Map<SomePropertyOfMyObject, Consumer<MyObject>>` and setup all different cases as lambdas where the parameter is the object being tested.
+* Prefix Interfaces with `I` - like `ITemplate`.
+* `@SupressWarnings("...")` should be avoided in General:
+  * If it needs to be applied, limit it to the smallest possible scope (and create a new variable for it if necessary).
+  * An Suppression should always have a comment stating why it is valid to suppress this warning here.
+  * In general only a `@SupressWarning("unchecked")` should ever be necessary - do not suppress deprecations!
+* Capitalize comments.
+* Use multiline comments where applicable, and single-line-comments only for single line comments.
+* Write Javadoc instead of comments, where possible:
+  * `@implNote` and `@implSpec` are always possible...
+  * Try to list all unchecked exceptions with `@throws`
+  * Leave uncommented parameters out (no `@param obj` without description)
+* Avoid giant Method-Bodies where possible, by splitting into smaller Methods
 * Use Guava's `Preconditions` class for Argument validation
 * Validate `@NonNull` values with `Objects.requireNonNull` if necessary
 * use getters and setters, even in the declaring class
 * Use protected fields in non-API superclasses, rather than private ones with getters and setters, if:
-      1. Those field are barely accessed in that superclass and are heavily accessed in its subclasses.
-      2. There is no additional need for getters and setters, such as:
-          1. External access to the field.
-          2. The need to run some additional code or logic when getting or setting the value of the field.
+  1. Those field are barely accessed in that superclass and are heavily accessed in its subclasses.
+  2. There is no additional need for getters and setters, such as:
+    1. External access to the field.
+    2. The need to run some additional code or logic when getting or setting the value of the field.
```

Comment:
Do you mean make them a single list?

Reply:
No, they are supposed to be 1, 2 (1, 2) (appearing as i, ii (a, b)), but they now appear as i-iv, since you reduced the number of spaces to indent them.

Reply:
See [here](https://github.com/Direwolf20-MC/BuildingGadgets/pull/268#issuecomment-473595074) for the current problem.
Just a extra note that the web editor marks the double indented entries red even if they show up "correctly" (the way we wanted) .

Reply:
Increasing the number of indenting spaces for lines 78-79 to 3 fixes those two lines. I don't care how many spaces are used where, as long as things actually function properly.

Reply:
If it's better to use 4 spaces everywhere, fine. If not, fine. Just make it function properly.